server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/GrammarPreprocessorImpl.java

```diff
@@ -66,9 +76,22 @@ public GrammarPreprocessorImpl(GrammarPreprocessorListenerFactory listenerFactor
 
     ParseTreeWalker walker = new ParseTreeWalker();
     GrammarPreprocessorListener listener =
-        listenerFactory.create(uri, tokens, copybookStack, copybookProcessingMode, recursiveReplaceStmtStack);
+        listenerFactory.create(
+            uri, tokens, copybookStack, copybookProcessingMode, recursiveReplaceStmtStack, replacingClauses);
     walker.walk(listener, startRule);
-
+    listener.getErrors().addAll(replaceListener.getErrors());
```

Comment:
I don't think changing the state of the listener is a good idea, it would be better to add a new list here to accumulate errors